core/context.go

```diff
@@ -26,17 +26,30 @@ import (
 	"go.uber.org/fx/service"
 )
 
-// Context embedds Host and go context for use
+// Context embeds Host and Go stdlib context for use
 type Context interface {
 	gcontext.Context
 	service.Host
+	WithContext(ctx gcontext.Context) *context
 }
 
 type context struct {
 	gcontext.Context
 	service.Host
 }
 
+// WithContext returns a shallow copy of c with its context changed to ctx.
+// The provided ctx must be non-nil. Follows from net/http Request WithContext.
+func (c *context) WithContext(ctx gcontext.Context) (newC *context) {
```

Comment:
well, the only thing i see is `WithContext` that can be changed to `With`, but other than that not much to be done :(